### PR TITLE
Support Gemini Pro as an option

### PR DESCRIPTION
**Pull request by pradh**
Drop the currently unused PaLM text API support, and replace it with Gemini Pro.

Rename "PALM" things with "LLM", since we now have both PaLM and Gemini models.

**Testing**

* Manually tested with `llm_api=geminipro&detector=llm` (example, http://localhost:8080/explore/#q=income+in+california+counties+with+median+age+over+40&detector=llm&llm_api=geminipro)

* Need to run adversarial with `llm_api=geminipro`

**Notes**

Generally, Gemini seems to adhere to the prompt better.  For instance, [employment where obama was born], PaLM will expand the place to Hawaii, but Gemini won't.  Gemini also does better with 2 variable constraint queries, like [income in california counties with median age over 40], while PaLM often gets it wrong.

The Gemini API is also faster than PaLM Chat (the one we use).

### CODE REVIEW

File: server/integration_tests/standalone/adversarial.py [hunk 1]
```diff
@@ -52,8 +52,8 @@ class Mode:
 
 flags.DEFINE_enum(
     "llm_api",
-    "chat",
-    ["chat", "text"],
+    "palm",
+    ["palm", "geminipro"],
     f"Value for the llm_api parameter. Valid values: chat, text.",
```
Comment (chejennifer) [inline server/integration_tests/standalone/adversarial.py:57]: update valid values here?

File: server/lib/nl/detection/detector.py [hunk 2]
```diff
@@ -65,13 +65,13 @@ def detect(
   #
   # In the absence of the PALM API key, fallback to heuristic.
   #
-  if (detector_type in _PALM_API_DETECTORS and
-      'PALM_API_KEY' not in current_app.config):
+  if (detector_type in _LLM_API_DETECTORS and
+      'LLM_API_KEY' not in current_app.config):
     counters.err('failed_palm_keynotfound', '')
```
Comment (chejennifer) [inline server/lib/nl/detection/detector.py:70]: should we update this to 'failed_llm_keynotfound'?

File: server/lib/nl/detection/detector.py [hunk 3]
```diff
@@ -65,13 +65,13 @@ def detect(
...
     detector_type = RequestedDetectorType.Heuristic.value
 
-  if (detector_type in _PALM_API_DETECTORS and
-      'PALM_PROMPT_TEXT' not in current_app.config):
+  if (detector_type in _LLM_API_DETECTORS and
+      'LLM_PROMPT_TEXT' not in current_app.config):
     counters.err('failed_palm_promptnotfound', '')
```
Comment (chejennifer) [inline server/lib/nl/detection/detector.py:75]: should we update this to 'failed_llm_promptnotfound'?

File: server/lib/translator.py [hunk 4]
```diff
@@ -168,9 +168,9 @@ def _maybe_add(string: str):
 
 def _translate(queries: list[str], source_lang: str,
                target_lang: str) -> list[str]:
-  # The name "PALM_API_KEY" is a misnomer.
+  # The name "LLM_API_KEY" is a misnomer.
   # Both palm and translation apis are enabled under this key.
```
Comment (chejennifer) [inline server/lib/translator.py:172]: palm should be llm?

File: server/tests/lib/nl/detection/llm_api_test.py [hunk 5]
```diff
@@ -106,16 +106,19 @@ class TestParseResponse(unittest.TestCase):
   def test_chat(self, input, want):
```
Comment (chejennifer) [inline server/tests/lib/nl/detection/llm_api_test.py:106]: should test_chat and test_text_block be updated to test_gemini and test_palm? and then use the respective extraction functions in each function?

**Review by chejennifer — COMMENTED** [review]
nice! Just a few places that I think need a naming update

File: server/tests/lib/nl/detection/llm_api_test.py (hunk 5 above)
Comment (pradh) [reply server/tests/lib/nl/detection/llm_api_test.py]: test_chat has both palm and gemini, but renamed them slightly.

File: server/lib/translator.py (hunk 4 above)
Comment (pradh) [reply server/lib/translator.py]: yes!

File: server/lib/nl/detection/detector.py (hunk 3 above)
Comment (pradh) [reply server/lib/nl/detection/detector.py]: yes!

File: server/integration_tests/standalone/adversarial.py (hunk 1 above)
Comment (pradh) [reply server/integration_tests/standalone/adversarial.py]: yes!

File: server/lib/nl/detection/detector.py (hunk 2 above)
Comment (pradh) [reply server/lib/nl/detection/detector.py]: yes!

**Review by pradh — COMMENTED** [review]
Thanks for the careful review!

**Review by chejennifer — APPROVED** [review]
thanks for the updates!

**Pull request merged**